src/Includes/Shader.hpp

```diff
@@ -80,6 +80,12 @@ class ShaderProgram {
   void setUniform(const std::string& name, float val);
   void setUniform(const std::string& name, int val);
 
```

Comment:
I just have setUniform(name, type) for all these above. That way I don't have to do setUniformType(name, type).